plugins/com.google.cloud.tools.eclipse.appengine.login/src/com/google/cloud/tools/eclipse/appengine/login/ui/LoginCredentialExporter.java

```diff
@@ -0,0 +1,49 @@
+package com.google.cloud.tools.eclipse.appengine.login.ui;
+
+import java.io.IOException;
+import java.nio.file.Files;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.jface.window.IShellProvider;
+
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.cloud.tools.eclipse.appengine.login.CredentialHelper;
+import com.google.cloud.tools.eclipse.appengine.login.GoogleLoginService;
+import com.google.cloud.tools.eclipse.util.status.StatusUtil;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Charsets;
+
+public class LoginCredentialExporter {
+
+  private static final String CREDENTIAL_FILENAME = "gcloud-credentials.json";
+  
+  private GoogleLoginService loginService;
+  private CredentialHelper credentialHelper;
+  
+  public LoginCredentialExporter() {
+    this(new GoogleLoginService(), new CredentialHelper());
+  }
+  
+  @VisibleForTesting
+  LoginCredentialExporter(GoogleLoginService googleLoginService, CredentialHelper credentialHelper) {
+    this.loginService = googleLoginService;
+    this.credentialHelper = credentialHelper;
+  }
+
+  public void logInAndSaveCredential(IPath workDirectory, IShellProvider shellProvider) throws IOException, 
```

Comment:
Let's add a comment

`Must be called from a UI context.`

because this is a public-facing utility class being exported to other bundles.